File: 02_classes_metodos/elixir/ex03.ex
defmodule Cofre do
  def abrir(senha) do
    if senha_valida?(senha) do
      IO.puts("Cofre aberto!")
    else
      IO.puts("Senha Invalida")
    end
  end

  defp senha_valida?(senha) do
    pass = to_string(senha)

    if pass == "1234" do
      true
    else
      false
    end
  end
end

Cofre.abrir(1234)
Cofre.abrir(12345)

File: 02_classes_metodos/ruby/ex03.rb
class Cofre
  def abrir(senha)
    if senha_valida?(senha)
      puts "Cofre Aberto!"
    else
      puts "Senha Invalida."
    end
  end

  private

  def senha_valida?(senha)
    pass = senha.to_s()

    if pass == "1234"
      true
    else
      false
    end
  end
end

if __FILE__ == $0
  cofre = Cofre.new
  cofre.abrir(1234)
  cofre.abrir(12345)
end